csrc/preseg_passes/fmin_fmax_promotion.cpp

```diff
@@ -0,0 +1,234 @@
+
+  valMap[in_tv] = ValStatus::DEFAULT;
+  valMap[out_tv] = ValStatus::BAD;
+
+  auto traversal =
+      StmtSort::getExprsBetween({targetRop->input(0)}, fusion->outputs());
+
+  for (Expr* expr : traversal) {
+    std::string opName = expr->getOpString();
+
+    if (expr == targetRop) {
+      // Skip the target rop. We already marked its status.
+      continue;
+    }
+
+    bool anyBad = AnyBadInputTvs(expr, valMap);
+    bool anyBadDefault = AnyBadDefaultInputTvs(expr, valMap);
+    bool anyDefault = AnyDefaultInputTvs(expr, valMap);
+
+    auto* out_tv = dynamic_cast<TensorView*>(expr->output(0));
+
+    bool canBeAnalyzed = expr->isA<UnaryOp>() || expr->isA<ReductionOp>() ||
+        expr->isA<BroadcastOp>() || expr->isA<BinaryOp>();
+
+    if (!out_tv || !canBeAnalyzed) {
+      if (anyBad) {
+        return false;
+      } else {
+        continue;
+      }
+    }
+
+    if (AnyGoodInputTvs(expr, valMap)) {
+      valMap[out_tv] = ValStatus::GOOD;
+      continue;
+    }
+
+    bool mappedReduction = false;
+    if (expr->isA<ReductionOp>()) {
+      if (valMap[expr->input(0)->as<TensorView>()] == ValStatus::DEFAULT ||
+          valMap[expr->input(0)->as<TensorView>()] == ValStatus::BAD_DEFAULT) {
+        for (IterDomain* out_id : out_tv->getLogicalDomain()) {
+          if (out_id->isReduction()) {
+            if (logical_map.canMap(
+                    targetRop->input(0)->as<TensorView>()->domain(),
+                    reduceIn,
+                    out_tv->domain(),
+                    out_id)) {
+              valMap[out_tv] = ValStatus::GOOD;
+              mappedReduction = true;
+              break;
+            }
+          }
+        }
+      }
+    }
+    if (mappedReduction) {
+      continue;
+    }
+
+    if (anyBadDefault) {
+      valMap[out_tv] = ValStatus::BAD_DEFAULT;
+      continue;
+    }
+
+    if (anyDefault && anyBad) {
+      valMap[out_tv] = ValStatus::BAD_DEFAULT;
+      continue;
+    }
+
+    if (anyDefault) {
+      valMap[out_tv] = ValStatus::DEFAULT;
+      continue;
+    }
+
+    if (anyBad) {
+      valMap[out_tv] = ValStatus::BAD;
+    }
+  }
+
+  // Check whether any bad status reached output nodes
+  auto output_tvs = ir_utils::filterByType<TensorView>(fusion->outputs());
+  for (TensorView* out_tv : output_tvs) {
+    if (valMap[out_tv] == ValStatus::BAD ||
+        valMap[out_tv] == ValStatus::BAD_DEFAULT) {
+      return false;
+    }
+  }
+
+  return true;
+}
+
+bool AnalyzeMinMaxOp(ReductionOp* targetRop) {
+  auto* in_tv = targetRop->input(0)->as<TensorView>();
+  auto* out_tv = targetRop->output(0)->as<TensorView>();
+
+  ComputeAtLogicalDomainMap logical_map;
+  logical_map.build(true);
+
+  auto c2p = logical_map.mapBestEffort(
+      out_tv->domain(),
+      out_tv->getLogicalDomain(),
+      in_tv->domain(),
+      in_tv->getLogicalDomain());
+
+  for (IterDomain* out_id : out_tv->getLogicalDomain()) {
+    if (out_id->isReduction()) {
+      IterDomain* in_id = c2p[out_id];
```

Comment:
Use `.at` in order to throw an error if `out_id` is not found.
```suggestion
      IterDomain* in_id = c2p.at(out_id);
```